lib/Reader/abi.cpp

```diff
@@ -29,9 +29,9 @@
 using namespace llvm;
 
 // Static class with helpers for the Microsoft x86-64 ABI.
-class X86_64_Win64 {
+class XArchWin64 {
```

Comment:
We can exclude abi code from clang-tidy checks. 

The other option is to modify the (MS version of) clang-tidy to make an exception for the x86_64 convention -- which is not a super-clean solution.